synapse/lib/storm.py

```diff
@@ -101,13 +101,52 @@ def getArgParser(self):
 
     def runStormCmd(self, snap, genr):
 
-        count = 0
+        for count, item in enumerate(genr):
 
-        for item in genr:
-
-            yield item
-
-            count += 1
             if count >= self.opts.count:
                 snap.printf(f'limit reached: {self.opts.count}')
                 break
+
+            yield item
+
+class DelNodeCmd(Cmd):
+    '''
+    Delete nodes produced by the previous query logic.
+
+    (no nodes are returned)
+
+    Example
+
+        inet:fqdn=vertex.link | delnode
+    '''
+    name = 'delnode'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('--sudo', help='Activate elevated admin privs if allowed.')
+        return pars
+
+    def runStormCmd(self, snap, genr):
+
+        # a bit odd, but we need to be detected as a generator
+        yield from ()
+
+        if self.opts.sudo:
+            snap.elevated = True
+
+        for node in genr:
+            node.delete()
```

Comment:
Do we want to yield a message indicating the success of deleting a node?

Reply:
It would be nice to know which nodes were deleted

Reply:
The node:del splice should fulfill this purpose.

Reply:
Maybe even a message like "N nodes deleted"